MSAL/test/unit/MSALExternalAccountHandlerTests.m

```diff
@@ -65,29 +66,42 @@ - (BOOL)removeAccount:(id<MSALAccount>)account
        tenantProfiles:(nullable NSArray<MSALTenantProfile *> *)tenantProfiles
                 error:(NSError * _Nullable * _Nullable)error
 {
-    self.removeAccountInvokedCount++;
+    return [self removeAccount:account
+                   wipeAccount:NO
+                tenantProfiles:tenantProfiles
+                         error:error];
+}
+
+- (nullable NSArray<id<MSALAccount>> *)accountsWithParameters:(MSALAccountEnumerationParameters *)parameters
+                                                        error:(NSError * _Nullable * _Nullable)error
+{
+    self.readAccountsInvokedCount++;
     
     if (self.accountOperationError && error)
     {
         *error = self.accountOperationError;
     }
     
-    return self.accountOperationResult;
+    return self.resultAccounts;
```

Comment:
I am not sure how self.resultAccounts is created and whether it can have some objects even if we encountered self.accountOperationError. To be safe, can we return nil in the if statement to avoid the possibility of returning both self.resultAccounts and filling the error object.

Reply:
Sounds good, will address.

Reply:
Addressed.